Added bazelisk

sluongng opened the pull request:
When renovate is used together with Bazel repositories, there may be
some certain Bazel targets needed to be run for each dependencies
upgrade.

A good example would be upgrading go.mod file inside a Bazel-Gazelle
setup would require Gazelle to be run after go.mod and go.sum update to
also update the dependencies tracked inside WORKSPACE files (or linked
starlark files).

Add bazelisk into the toolchain so that renovate-full image could be
used for such situations.

Close https://github.com/renovatebot/docker-buildpack/issues/51

---

Small note:
I am not too familiar with the structure and usage of this repo. Looking forward for pointers from the maintainers.

JamieMagee commented on test/bazel/Dockerfile:
```diff
@@ -0,0 +1,8 @@
+ARG IMAGE=renovate/buildpack
+FROM ${IMAGE} as build
+
+RUN install-tool bazelisk 1.7.4
```
This allows renovate to use the latest release of bazelisk for testing

```suggestion
# renovate: datasource=github-releases lookupName=bazelbuild/bazelisk
RUN install-tool bazelisk 1.7.4
```

viceice commented on src/bazel/build/bazel.sh:
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
```
move to `src/bazel/buildpack/tools/bazel.sh`

viceice reviewed (changes requested):
make non-root install compatible

See python for sample:
https://github.com/renovatebot/docker-buildpack/blob/8ee1e7ea600c7c2e29b25aca1a54c7c28c31c5ab/src/python/buildpack/tools/python.sh

sluongng commented:
Updated the MR:

- Rebase on latest origin/main
- Upgrade bazelisk version to 1.7.5
- Added the suggested Renovate version tracking
- Moved script to buildpack dir to make it non-root compatible

@JamieMagee @viceice  could you please give this another round of review?

sluongng commented:
I also removed `bazel` from the github action build matrix.

My understanding is that it uses a `target-{matrix-target}` image as base image

```
#--------------------------------------
# Image: final
#--------------------------------------
FROM target-${TARGET} as final

ENTRYPOINT ["docker-entrypoint.sh"]
CMD ["bash"]
```

And because there is not yet a `bazel` image for this, it failed.

Not sure if this needed to be build manually or with a separate PR post-merge?

viceice commented:
Do we need a `renovate/bazel` image? Or is it only to install to the renovate full image / custom renovate image ?

sluongng commented:
> Do we need a `renovate/bazel` image? Or is it only to install to the renovate full image / custom renovate image ?

For my use case I will only need it inside the full image as that's how I am running the bot (without dind). Not sure about other folks.

Perhaps let's start with having it only in the full image? If others have a need to have it in container then it's 1 PR away 🤗

sluongng commented:
@viceice @JamieMagee  is there specific items you guys want me to address here before merging?

viceice commented on src/bazel/buildpack/tools/bazel.sh:
```diff
@@ -0,0 +1,26 @@
+fi
+
+# bazel depends on
+apt_install git diff patch
```
Git should already be installed, do we really should try again? 🤔

viceice reviewed (commented):
There should be a full bazelisk sample call,  you can add it to `test/latest`

viceice commented:
Please open a new pr at https://github.com/containerbase/buildpack
